components/dotenv.rst

```diff
@@ -67,6 +67,42 @@ The ``load()`` method never overwrites existing environment variables. Use the
 .. versionadded:: 4.2
     The ``Dotenv::overload()`` method was introduced in Symfony 4.2.
 
+Symfony follows and promotes the `app config recommendations`_ made by the
+Twelve-Factor App methodology. Among other things, it's recommended to not group
+env vars into environments (``dev``, ``prod``, ``test``, etc.)
+
+When deploying apps you should follow that recommendation strictly. However,
+when developing apps on your local machine, testing multiple apps in different
+environments with regular env vars is cumbersome. That's why Symfony provides a
+feature to define env vars per environment using the ``loadForEnv()`` method
+(but you should only use it on your local machine, not in production servers)::
+
+    // ...
+    $dotenv->loadForEnv('dev', __DIR__.'/.env');
+    // Looks for env vars in these files (and in this order):
+    //   .env.dev.local
+    //   .env.local
+    //   .env.dev
+    //   .env
+
+    // ...
+    $dotenv->loadForEnv('test', __DIR__.'/.env');
+    // In the 'test' environment, the '.env.local' file is ignored. This example
+    // looks for env vars in these files (and in this order):
+    //   .env.test.local
+    //   .env.test
+    //   .env
+
```

Comment:
```suggestion
Similarly to ``DotEnv::load()``, ``DotEnv::loadForEnv()`` never overwrites existing environment variables, nor variables defined by files loaded previously. For instance, if the same variable is defined in both  `.env` and `.env.test.local`, the value will be the one defined in `.env.test.local`, because this file is loaded first by ``DotEnv::loadForEnv()``.
```

Reply:
I would add somewhere something similar to https://github.com/bkeepers/dotenv/blob/master/README.md#can-i-use-dotenv-in-production last paragraph:

> please note that env vars that are general to all environments should be stored in .env. Then, environment specific env vars should be stored in .env.<that environment's name>.

i.e no need to repeat all env vars. Only the ones likely to be specific for the target env.